dask/dataframe/io/parquet/arrow.py

```diff
@@ -243,7 +243,7 @@ def _collect_pyarrow_dataset_frags(
                 "partition_keys": {},
                 "partition_names": [],
             }
-            return metadata, partition_info
+            return sorted(metadata, key=natural_sort_key), partition_info
```

Comment:
This seem fine, though I'm wondering why it's needed. Is there a test we can add?

Reply:
This "fix" is the purpose of the PR, and the modified test is meant to cover this exact change.

Pyarrow does not produce a list of fragments that are sorted by path name.  We sort the fragments by path name in other places, but missed this particular "fast path" until now.

Reply:
Wow, I didn't register that you added a test for this even though I commented on it 🤦  Thanks for clarifying